File: frontend/demo/component/basiclayouts/basic-layouts-expanding-items.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/horizontal-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('basic-layouts-expanding-items')
export class Example extends LitElement {
  connectedCallback() {
    super.connectedCallback();
    this.classList.add('basic-layouts-example');
  }

  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-horizontal-layout theme="padding spacing">
        <div class="example-item" style="flex-grow: 1">Item 1</div>
        <div class="example-item">Item 2</div>
        <div class="example-item">Item 3</div>
      </vaadin-horizontal-layout>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/basiclayouts/BasicLayoutsExpandingItems.java
package com.vaadin.demo.component.basiclayouts;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("basic-layouts/expanding-items")
public class BasicLayoutsExpandingItems extends Div {

    public BasicLayoutsExpandingItems() {
        // tag::snippet[]
        Div item1 = new Div("Item 1");
        item1.setClassName("layout-item");
        HorizontalLayout layout = new HorizontalLayout();
        layout.setFlexGrow(1, item1);
        // end::snippet[]
        layout.setPadding(true);
        layout.add(item1);
        
        Div item2 = new Div("Item 2");
        item2.setClassName("layout-item");
        layout.add(item2);
        
        Div item3 = new Div("Item 3");
        item3.setClassName("layout-item");
        layout.add(item3);
        
        this.add(layout);
        this.setClassName("basic-layouts-example");
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<BasicLayoutsExpandingItems> { // hidden-source-line
    } // hidden-source-line
}
